packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -0,0 +1,268 @@
+
+  const template = `
+  import { ethers, EventFilter, Signer, BigNumber, BigNumberish, PopulatedTransaction } from 'ethers';
+  import { ${contractImports.join(', ')} } from '@ethersproject/contracts';
+  import { BytesLike } from '@ethersproject/bytes';
+  import { Listener, Provider } from '@ethersproject/providers';
+  import { FunctionFragment, EventFragment } from '@ethersproject/abi';
+
+  interface ${contract.name}Interface extends ethers.utils.Interface {
+    functions: {
+      ${values(contract.functions)
+        .map((v) => v[0])
+        .map(generateInterfaceFunctionDescription)
+        .join('\n')}
+    };
+
+    events: {
+      ${values(contract.events)
+        .map((v) => v[0])
+        .map(generateInterfaceEventDescription)
+        .join('\n')}
+    };
+  }
+
+  export class ${contract.name} extends Contract {
+    connect(signerOrProvider: Signer | Provider | string): this;
+    attach(addressOrName: string): this;
+    deployed(): Promise<this>;
+
+    on(event: EventFilter | string, listener: Listener): this;
+    once(event: EventFilter | string, listener: Listener): this;
+    addListener(eventName: EventFilter | string, listener: Listener): this;
+    removeAllListeners(eventName: EventFilter | string): this;
+    removeListener(eventName: any, listener: Listener): this;
+
+    interface: ${contract.name}Interface;
+
+    functions: {
+      ${values(contract.functions)
+        .map(codegenFunctions.bind(null, { returnResultObject: true }))
+        .join('\n')}
+    };
+
+    ${values(contract.functions).map(codegenFunctions.bind(null, {})).join('\n')}
+
+    staticCall: {
+      ${values(contract.functions)
+        .map(codegenFunctions.bind(null, { isStaticCall: true }))
+        .join('\n')}
+    };
+
+    filters: {
+      ${values(contract.events)
+        .map((v) => v[0])
+        .map(generateEvents)
+        .join('\n')}
+    };
+
+    estimateGas: {
+      ${values(contract.functions)
+        .map((v) => v[0])
+        .map(generateEstimateFunction)
+        .join('\n')}
+    };
+
+    populateTransaction: {
+      ${values(contract.functions)
+        .map((v) => v[0])
+        .map(generatePopulateTransactionFunction)
+        .join('\n')}
+    };
+  }`
+
+  return template
+}
+
+export function codegenContractFactory(contract: Contract, abi: any, bytecode?: BytecodeWithLinkReferences): string {
+  const constructorArgs =
+    (contract.constructor && contract.constructor[0] ? generateInputTypes(contract.constructor[0].inputs) : '') +
+    `overrides?: ${contract.constructor[0]?.stateMutability === 'payable' ? 'PayableOverrides' : 'Overrides'}`
+  const constructorArgNamesWithoutOverrides =
+    contract.constructor && contract.constructor[0] ? generateParamNames(contract.constructor[0].inputs) : ''
+  const constructorArgNames = constructorArgNamesWithoutOverrides
+    ? `${constructorArgNamesWithoutOverrides}, overrides || {}`
+    : 'overrides || {}'
```

Comment:
does the `ContractFactory` require passing empty overrides, since you added the explicit `|| {}` here?

Reply:
When we don't pass any override using typechain's overrides, the `overrides` is set as `undefined`. But it looks like ethers v4 was fine with undefined `overrides` value but in v5 it throws an error. We can make it not pass an undefined override (+1 or 2 lines) or just fall-backing to an empty override as done corrently. Is it good doing like this?

Reply:
This is weird, not passing and optional argument _should_ be handled exactly the same as explicitly passing `undefined` there. TBH, if passing `undefined` overrides throws in ethers v5, it's likely a bug there. We can leave the explicit empty overrides for now.